app/src/main/java/de/westnordost/streetcomplete/data/osm/split/SplitWayAtPosition.kt

```diff
@@ -0,0 +1,28 @@
+package de.westnordost.streetcomplete.data.osm.split
+
+import de.westnordost.osmapi.map.data.Node
+import de.westnordost.osmapi.map.data.Way
+
+/** data class that carries the information for one split to perform on a random position on a way. */
+class SplitWayAtPosition(way: Way, val firstNode: Node, val secondNode: Node, val delta: Double) {
```

Comment:
Not sure where it should be handled - maybe here, maybe in upper levels of API that are not yet written.

But in general, especially in StreetComplete delta extremely close to 0 or 1 should be clamped to 0/1.

It makes no sense to split way and create a new node centimeters from already existing one - it is better to split way on an already existing node.

Reply:
It should be handled in the UI as the clamp should be visible there as well